Engine/DataFeeds/Enumerators/Factories/BaseDataSubscriptionEnumeratorFactory.cs

```diff
@@ -68,16 +68,20 @@ public BaseDataSubscriptionEnumeratorFactory(bool isLiveMode, Func<SubscriptionR
         /// <returns>An enumerator reading the subscription request</returns>
         public IEnumerator<BaseData> CreateEnumerator(SubscriptionRequest request, IDataProvider dataProvider)
         {
+            // We decide to use the ZipDataCacheProvider instead of the SingleEntryDataCacheProvider here
+            // for resiliency and as a fix for an issue preventing us from reading non-equity options data.
+            // It has the added benefit of caching any zip files that we request from the filesystem, and reading
+            // files contained within the zip file, which the SingleEntryDataCacheProvider does not support.
             var sourceFactory = request.Configuration.GetBaseDataInstance();
-
-            using (var dataCacheProvider = new SingleEntryDataCacheProvider(dataProvider))
+            using (var dataCacheProvider = new ZipDataCacheProvider(dataProvider))
```

Comment:
I think this will have other implications later but I can't remember them precisely. This might have been done to reduce the RAM used?